controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -252,15 +252,21 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, kcp *cont
 	case numMachines < desiredReplicas && numMachines == 0:
 		// create new Machine w/ init
 		logger.Info("Scaling to 1", "Desired Replicas", desiredReplicas, "Existing Replicas", numMachines)
-		if initErr := r.initializeControlPlane(ctx, cluster, kcp, logger); initErr != nil {
+		if initErr := r.initializeControlPlane(ctx, cluster, kcp); initErr != nil {
```

Comment:
```suggestion
		if err := r.initializeControlPlane(ctx, cluster, kcp); err != nil {
```

Nit

Reply:
I haven't been overly consistent about it, but I used `initErr` here to avoid shadowing the `err` that was defined previously. I've been trying to avoid shadowing vars unless there is a particular need to.

Reply:
I suggested that because the error is only defined within the `if` scope anyway, and it'll be unlikely that it'll need to be looking at any other error outside that scope

Reply:
fair enough.